src/components/game/Sketch.tsx

```diff
@@ -198,6 +85,55 @@ const SketchComponent = ({ game, evaluation }: SketchProps) => {
     boardGroundPath && sketch.image(imageFromPath(boardGroundPath), 0, 0, sketch.width, sketch.height)
   }
 
+  const loadedSounds: Map<Id, { lastStatus: string, soundFile: p5.SoundFile, started: boolean }> = new Map()
+
+  function playSounds() {
+    currentSoundStates(evaluation).forEach((soundState: SoundState) => {
+      if (!loadedSounds.has(soundState.id)) {
+        loadedSounds.set(soundState.id,
+          {
+            lastStatus: soundState.status,
+            soundFile: new p5.SoundFile(game.assetsDir + soundState.file),
+            started: false,
+          })
+      }
+
+      const loadedSound = loadedSounds.get(soundState.id)!
+      loadedSound.soundFile.setLoop(soundState.loop)
+      loadedSound.soundFile.setVolume(soundState.volume)
+
+      if ((loadedSound.lastStatus !== soundState.status && loadedSound.soundFile.isLoaded()) || (!loadedSound.started && loadedSound.soundFile.isLoaded())) {
+        loadedSound.started = true
+
+        switch (soundState.status) {
+          case "played": {
+            loadedSound.soundFile.play()
+            break
+          }
+          case "paused": {
+            loadedSound.soundFile.pause()
+            break
+          }
+          case "stopped": {
+            loadedSound.soundFile.stop()
+          }
+        }
+      }
+
+      loadedSound.lastStatus = soundState.status
+
+    })
+  }
+
+  function removeUnusedLoadedSounds() {
+    const soundIdsBeingUsed: Id[] = currentSoundStates(evaluation).map((soundState: SoundState) => soundState.id)
+    const unusedSoundIds: Id[] = [...loadedSounds.keys()].filter((id: Id) => !soundIdsBeingUsed.includes(id))
```

Comment:
Creo que esta deconstrucción de la lista no es necesaria:

```suggestion
    const unusedSoundIds: Id[] = loadedSounds.keys().filter((id: Id) => !soundIdsBeingUsed.includes(id))
```

Reply:
El `loadedSounds.keys()` devuelve un `IterableIterator` asi que creo que hay hacerlo asi para tratarlo como una lista

Reply:
Okkk!